### ISSUE TITLE
text options appear to be broken

### ISSUE THREAD

**Issue by GrumpyLittleTed** [spencermountain/compromise#953]
In the example from https://observablehq.com/@spencermountain/compromise-text
the result is not as expected:

{
  let doc = nlp(`Dr. John Smith-McDonald...`)
  let opts = {
    punctuation: true,
    abbreviations: true,
    case: true,
  }
  return doc.text(opts)
}

result is "Dr.` John Smith-McDonald..." (unchanged from input text)

**Comment by spencermountain** [comment]
thanks, not sure what's up here. will check it out for next release

**Comment by spencermountain** [comment]
oof, these text options are a mess, and need to be fixed in our next breaking release.

Right now the fix on dev now supports this, will release it tmrw:
```js
let doc = nlp("Dr. John Smith-McDonald...?  ")
let opts = {
  keepPunct: false,
  keepSpace: false,
  case: false,
}
console.log(doc.text(opts) + '|')
// 'Dr. John Smith McDonald|'
```
cheers

**Comment by spencermountain** [comment]
released in 14.7.0
thanks
